build-settings/rollup.config.js

```diff
@@ -1,33 +1,56 @@
 /* eslint-disable import/no-commonjs */
 import fs from "fs";
+import path from "path";
 import autoExternal from "rollup-plugin-auto-external";
 import {babel} from "@rollup/plugin-babel";
 import {terser} from "rollup-plugin-terser";
 import copy from "rollup-plugin-copy";
+import resolve from "@rollup/plugin-node-resolve";
+import replace from "@rollup/plugin-replace";
 
-const {presets, plugins} = require("./babel.config.js");
+const createBabelPresets = require("./create-babel-presets.js");
 
 /**
- * Creates the rollup config shared by all our output formats.
+ * Make path to a package relative path.
  */
-const createSharedConfig = (pkgName) => ({
-    input: `packages/${pkgName}/src/index.js`,
+const makePackageBasedPath = (pkgName, pkgRelPath) =>
+    path.normalize(path.join("packages", pkgName, pkgRelPath));
+
+/**
+ * Generate the rollup output configuration for a given
+ */
+const createOutputConfig = (pkgName, format, targetFile) => ({
+    file: makePackageBasedPath(pkgName, targetFile),
+    sourcemap: true,
+    format,
+});
+
+const createConfig = ({name, format, platform, file}) => ({
+    output: createOutputConfig(name, format, file),
+    input: makePackageBasedPath(name, "./src/index.js"),
     plugins: [
+        replace({
+            preventAssignment: true,
+            values: {
+                "process.env.NODE_ENV": JSON.stringify(
+                    process.env.NODE_ENV || "production",
+                ),
+            },
+        }),
```

Comment:
This enables us to do `process.env.NODE_ENV` checks during build; and rollup will omit code accordingly.

Reply:
Huh - rollup doesn't support this natively? That's surprising!

Reply:
Yeah. I based this off React's use of rollup. I was a little surprised too.